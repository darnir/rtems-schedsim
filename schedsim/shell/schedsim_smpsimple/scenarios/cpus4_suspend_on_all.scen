#
#  Check blocking self on secondary CPU
#
rtems_init 4
task_create smp1 5
task_create smp2 6
task_create smp3 7
task_create smp4 8
dispatch
# GOAL: want 0x0a010001 - 0x0a010004 as heir and executings
# GOAL:   on CPUs 0 - 3 respectively
cpus
task_suspend smp1
task_suspend smp2
task_suspend smp3
task_suspend smp4
dispatch
# GOAL: want IDLE 0x09010001 - 0x09010004 as heir and executings
# GOAL:   on CPUs 0 - 3 respectively
cpus
