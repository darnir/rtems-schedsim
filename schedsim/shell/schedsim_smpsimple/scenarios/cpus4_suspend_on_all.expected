Script File               : scenarios/cpus4_suspend_on_all.scen
verbose                   : 1
==> 1: #
==> 2: #  Check blocking self on secondary CPU
==> 3: #
==> 4: rtems_init 4
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
  CPU 1: Thread Heir: 0x09010002 priority=255
  CPU 1: Thread Executing: 0x09010002 priority=255
  CPU 2: Thread Heir: 0x09010003 priority=255
  CPU 2: Thread Executing: 0x09010003 priority=255
  CPU 3: Thread Heir: 0x09010004 priority=255
  CPU 3: Thread Executing: 0x09010004 priority=255
==> 5: task_create smp1 5
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=5
Task (smp1) starting: id=0x0a010001, priority=5
  CPU 3: Thread Heir: 0x0a010001 priority=5
  CPU 3: Thread Executing: 0x0a010001 priority=5
==> 6: task_create smp2 6
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=6
Task (smp2) starting: id=0x0a010002, priority=6
  CPU 2: Thread Heir: 0x0a010002 priority=6
  CPU 2: Thread Executing: 0x0a010002 priority=6
==> 7: task_create smp3 7
Creating task with:  Preemption: yes Timeslicing: no
Task (smp3) created: id=0x0a010003, priority=7
Task (smp3) starting: id=0x0a010003, priority=7
  CPU 1: Thread Heir: 0x0a010003 priority=7
  CPU 1: Thread Executing: 0x0a010003 priority=7
==> 8: task_create smp4 8
Creating task with:  Preemption: yes Timeslicing: no
Task (smp4) created: id=0x0a010004, priority=8
Task (smp4) starting: id=0x0a010004, priority=8
  CPU 0: Thread Heir: 0x0a010004 priority=8
  CPU 0: Thread Executing: 0x0a010004 priority=8
==> 9: # GOAL: want 0x0a010001 - 0x0a010004 as heir and executings
==> 10: # GOAL:   on CPUs 0 - 3 respectively
==> 11: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  8 / 0x0a010004 @  8          false
  CPU 1: 0x0a010003 @  7 / 0x0a010003 @  7          false
  CPU 2: 0x0a010002 @  6 / 0x0a010002 @  6          false
  CPU 3: 0x0a010001 @  5 / 0x0a010001 @  5          false
=== End of Ready Set of Threads
==> 12: task_suspend smp1
Suspending task (0x0a010001)
  CPU 3: Thread Heir: 0x09010001 priority=255
  CPU 3: Thread Executing: 0x09010001 priority=255
==> 13: task_suspend smp2
Suspending task (0x0a010002)
  CPU 2: Thread Heir: 0x09010002 priority=255
  CPU 2: Thread Executing: 0x09010002 priority=255
==> 14: task_suspend smp3
Suspending task (0x0a010003)
  CPU 1: Thread Heir: 0x09010003 priority=255
  CPU 1: Thread Executing: 0x09010003 priority=255
==> 15: task_suspend smp4
Suspending task (0x0a010004)
  CPU 0: Thread Heir: 0x09010004 priority=255
  CPU 0: Thread Executing: 0x09010004 priority=255
==> 16: # GOAL: want IDLE 0x09010001 - 0x09010004 as heir and executings
==> 17: # GOAL:   on CPUs 0 - 3 respectively
==> 18: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x09010004 @255 / 0x09010004 @255          false
  CPU 1: 0x09010003 @255 / 0x09010003 @255          false
  CPU 2: 0x09010002 @255 / 0x09010002 @255          false
  CPU 3: 0x09010001 @255 / 0x09010001 @255          false
=== End of Ready Set of Threads
